### PR TITLE
[Spark] Record an event on type widening

### PR DESCRIPTION
**Pull request by johanl-db**
## What changes were proposed in this pull request?
This change records an event `delta.typeWidening.typeChanges` whenever one or more type changes are applied to a Delta table.

For consistency, the existing event `delta.typeWideningFeatureRemovalMetrics` is renamed to `delta.typeWidening.featureRemovalMetrics`.

## How was this patch tested?
- Added a test covering event `delta.typeWidening.typeChanges`
- Added a test covering event `delta.typeWidening.featureRemovalMetrics`

### CODE REVIEW

File: spark/src/main/scala/org/apache/spark/sql/delta/PreDowngradeTableFeatureCommand.scala [hunk 1]
```diff
@@ -211,7 +211,7 @@ case class TypeWideningPreDowngradeCommand(table: DeltaTableV2)
 
     recordDeltaEvent(
       table.deltaLog,
-      opType = "delta.typeWideningFeatureRemovalMetrics",
+      opType = "delta.typeWidening.featureRemovalMetrics",
```
Comment (tomvanbussel) [inline spark/src/main/scala/org/apache/spark/sql/delta/PreDowngradeTableFeatureCommand.scala:214]: Nit: Just "delta.typeWidening.featureRemoval" or "delta.typeWidening.featureRemoval.metrics"

File: spark/src/main/scala/org/apache/spark/sql/delta/TypeWideningMetadata.scala [hunk 2]
```diff
@@ -126,18 +127,35 @@ private[delta] object TypeWideningMetadata {
...
         newField
     }
+
+    if (changesToRecord.nonEmpty) {
+      recordDeltaEvent(
+        deltaLog = txn.snapshot.deltaLog,
+        opType = "delta.typeWidening.typeChanges",
+        data = Map(
+          "changes" -> changesToRecord.map { change =>
+          Map(
```
Comment (tomvanbussel) [inline spark/src/main/scala/org/apache/spark/sql/delta/TypeWideningMetadata.scala:151]: Nit: indentation is a bit off here.

**Review by tomvanbussel — APPROVED** [review]
Two small nits, LGTM otherwise

File: spark/src/main/scala/org/apache/spark/sql/delta/PreDowngradeTableFeatureCommand.scala (hunk 1 above)
Comment (johanl-db) [reply spark/src/main/scala/org/apache/spark/sql/delta/PreDowngradeTableFeatureCommand.scala]: Changed to `delta.typeWidening.featureRemoval`

File: spark/src/main/scala/org/apache/spark/sql/delta/TypeWideningMetadata.scala (hunk 2 above)
Comment (johanl-db) [reply spark/src/main/scala/org/apache/spark/sql/delta/TypeWideningMetadata.scala]: Fixed

**Pull request merged**